src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1297,6 +1298,70 @@ void SolverInterfaceImpl::readScalarData(
   PRECICE_DEBUG("Read value = {}", value);
 }
 
+void SolverInterfaceImpl::setBoundingBoxes(
+    const int     meshID,
+    const double *boundingBox,
+    const int     size) const
+{
+  PRECICE_TRACE(meshID);
+  PRECICE_REQUIRE_MESH_USE(meshID);
+  if (size == 0)
+    return;
+  PRECICE_CHECK(boundingBox != nullptr, "The provided bounding boxes are empty.");
+  //  PRECICE_REQUIRE_MESH_MODIFY(meshID);
+  PRECICE_CHECK(size == 1, "Using more than one bounding box is not yet supported.");
+
+  // Get the related mesh
+  MeshContext & context = _accessor->meshContext(meshID);
+  mesh::PtrMesh mesh(context.mesh);
+  PRECICE_DEBUG("Define bounding box");
+  // Transform bounds into a suitable format
+  int                 dim = mesh->getDimensions();
+  std::vector<double> bounds(dim * 2);
+
+  for (int d = 0; d < dim; ++d) {
+    // Assert that min is lower or equal to max
+    PRECICE_CHECK(boundingBox[2 * d] <= boundingBox[2 * d + 1], "Your bounding box is ill defined, i.e. it has a negative volume. The required format is [x_min, x_max...]");
+    bounds[2 * d]     = boundingBox[2 * d];
+    bounds[2 * d + 1] = boundingBox[2 * d + 1];
+  }
+  // Create a bounding box
+  mesh::BoundingBox providedBoundingBox(bounds);
+  // Expand the mesh associated bounding box
+  mesh->expandBoundingBox(providedBoundingBox);
```

Comment:
So calling this multiple times will expand the bounding box.
Ideally, this would keep a collection of BBs, right?

This can be implement quite easily using a hierarchical data structure. (This is very common in physics.)

Reply:
Discussing with @uekerman: we decided to drop the collection idea here for the first steps. An extension might come in the future.